### PR TITLE
[IBS-68]:  Refactor Group Endpoints (staff)

### PR DESCRIPTION
**Pull request by DomiVesalius**
Resolves [IBS-68](https://ibs-utm.atlassian.net/browse/IBS-68?atlOrigin=eyJpIjoiNTIxZmQwY2IwNWU0NGQyYTk2ZjkyZWQwY2QyYmRmMzciLCJwIjoiaiJ9)

## Changes
- Refactored and created comprehensive tests for the following files:
    - `module/group/staff/add.js`
    - `module/group/staff/all.js`
    - `module/group/staff/check.js`
    - `module/group/staff/copy.js`
    - `module/group/staff/create.js`
    - `module/group/staff/extension.js`
    - `module/group/staff/remove.js`
- The refactoring of the files above also included refactoring of helper functions used within corresponding endpoints. Particularly the following functions in `utilities/helpers.js` were refactored/modified
    - `copy_groups`
    - `send_email_by_group`
- Also contains bugfixes to `MockVCS` class introduced by #66 

## Testing
Ensure tests work by running the following commands

```bash
npm test group_staff_add
npm test group_staff_all
npm test group_staff_check
npm test group_staff_copy
npm test group_staff_create
npm test group_staff_extension
npm test group_staff_remove
```

If #67 has been merged before you review, you can also run all tests automatically via the method described in the PR description

### Expected Outputs
image
image
image
image
image
image
image

### CODE REVIEW

File: backend/module/group/staff/all.js [hunk 1]
```diff
@@ -2,23 +2,38 @@ const express = require("express");
 const router = express.Router();
 const client = require("../../../setup/db");
 const helpers = require("../../../utilities/helpers");
```
Comment (hezhongq) [inline backend/module/group/staff/all.js:4]: Remove unused import statement

File: backend/module/group/staff/extension.js [hunk 2]
```diff
@@ -2,35 +2,32 @@ const express = require("express");
...
+	if (!group) return res.status(400).json({ message: "The group id doesn't exist." });
+
+	group.extension = req.body.extension;
+	await group.save();
+
+	res.status(200).json({ message: "The extension is changed." });
+
+	let subject = "IBS Extension Confirmation";
+	let body = "Your extension request for " + pg_res_update.rows[0]["task"] + " has been approved. The due date is extended by " + req.body["extension"] + " minutes. Please check IBS for details.";
+	helpers.send_email_by_group(res.locals["course_id"], req.body["group_id"], subject, body);
```
Comment (hezhongq) [inline backend/module/group/staff/extension.js:30]: May need to have await for the async function call

**Review by hezhongq — APPROVED** [review]
All tests passed and completed execution. Verified endpoint responses and message and they matched the format from before. Good test case coverage!

File: backend/module/group/staff/extension.js (hunk 2 above)
Comment (DomiVesalius) [reply backend/module/group/staff/extension.js:30]: I don't think `await` is required here as it is the last line of execution however I will include it for readability sake. 👍

**Pull request merged**